### PR TITLE
first draft of python-in-production-part-1-of-5

### PR DESCRIPTION
**Pull request by lward27**
Please review my blog post!

### CODE REVIEW

File: posts/python-in-production-part-1-of-5.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+To check which version of python you are running, type
+```bash
+python3 --version
+```
+
+For this series, we will be using Python version 3.10, although I will include some notes on how to get things working on earlier versions.  If you want to create a program to run on a different version of Pyton, you will need to make sure that you have that version of python installed, and that you create the environment using that version.
+
+To install Python 3.10 on a Mac, simply run the following command:
+```bash
+brew install python@3.10
```
Comment (lanefu) [inline posts/python-in-production-part-1-of-5.md:43]: I'd consider enlightening people on `pyenv`  if you're up for the extra deets

File: posts/python-in-production-part-1-of-5.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+Now that we have the proper software installed and a folder to house our project, lets create a virtual environment.
+
+**Python 3.3** or later (note: you will need to use what ever version of python you are building for in this next command, I will assume python3.10 from here on out):
+```bash
+python3.10 -m venv venv
+```
+
+**Python 3.2.6** or earlier:
```
Comment (lanefu) [inline posts/python-in-production-part-1-of-5.md:66]: would probably just ditch the older python stuff altogether

**Review by lanefu — APPROVED** [review]
nice work

File: posts/python-in-production-part-1-of-5.md (hunk 1 above)
Comment (lward27) [reply posts/python-in-production-part-1-of-5.md:43]: That would require my enlightenment as well. I will take a look!

File: posts/python-in-production-part-1-of-5.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+# Creating a Virtual Environment for Python
+
+The first step to creating a virtual environment requires us to think ahead, to the future.  What version of Python will this project ultimately run on? Will this project be a part of a legacy system running Python 2.7 or a cutting edge system running the latest version of Python 3?  Figuring this out ahead of time will make your life easier in the long run.  Once you have identified which version of Python you want to develop for, we need to make sure we have said version on our local machine.  If you are on a Mac, you likely already have Python installed. To check, simply run these two commands.
+
+To check which version of python you are running, type
+```bash
+python3 --version
+```
+
+For this series, we will be using Python version 3.10, although I will include some notes on how to get things working on earlier versions.  If you want to create a program to run on a different version of Pyton, you will need to make sure that you have that version of python installed, and that you create the environment using that version.
```
Comment (dsharpe) [inline posts/python-in-production-part-1-of-5.md:39]:
```suggestion
For this series, we will be using Python version 3.10, although I will include some notes on how to get things working on earlier versions.  If you want to create a program to run on a different version of Python, you will need to make sure that you have that version of python installed, and that you create the environment using that version.
```

File: posts/python-in-production-part-1-of-5.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+This can work for you, but if you have a lot of packages or dependencies, you may find it difficult to remember what you have already installed.  To make things even more messy, you run this next command only to find a list of every package you ever installed for every project you have worked on using this machine...
+```bash
+pip freeze
+```
+
+This nightmarish scenario can be avoided entirely by being intentional about your python code and the environment that it runs in.  With virtual environments you don't have to worry about overlapping dependencies, global dependencies, and you don't have to remember to type out which version of python3 you are running every time you need to install a package or run your program.
+
+# Managing Dependencies with Pip and a Virtual Environment
+
+If you have been following along, you should be in a directory named *sample-python-project* and have a folder named *venv* that contains your virtual environment.  Go ahead an *activate* the environment by sourcing it.
```
Comment (dsharpe) [inline posts/python-in-production-part-1-of-5.md:122]:
```suggestion
If you have been following along, you should be in a directory named *sample-python-project* and have a folder named *venv* that contains your virtual environment.  Go ahead and *activate* the environment by sourcing it.
```

File: posts/python-in-production-part-1-of-5.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+The command ***python3.10 -m pip install -r requirements.txt*** will install all of the requirements listed in that file.  To get our *boiler plate project* into git, simply go to github and create an empty repo named sample-python-project.  Once you have done that, we can get our local files into github by running a few simple commands.
+```bash
+git init
+git add -A
+git commit -m "initial commit"
+git remote add origin https://github.com/{{your_username}}/sample-python-project.git
+git push -u -f origin main
+```
+
+If you have been following along, then your directory listing should look something like this.
```
Comment (dsharpe) [inline posts/python-in-production-part-1-of-5.md:183]:
```suggestion
If you have been following along, then your directory listing should look something like this:
```

**Review by dsharpe — CHANGES_REQUESTED** [review]
Great article!

**Review by lward27 — COMMENTED** [review]
Thanks for the feedback!

**Review by dsharpe — APPROVED** [review]

**Pull request merged**